src/components/TextInput.jsx

```diff
@@ -114,32 +114,21 @@ export default class TextInput extends Component {
 
     if (!this.props.allowNewLine && value.match(newLineRegex) !== null) {
       value = event.target.value.replace(newLineRegex, ' ');
-
-      // controlled textarea must have value
```

Comment:
@jpuri Took me a while, but I believe I found a case where relying on `domNode.value = value;` is a problem.

The render function is executed whenever the state changes. Of course we can manipulate the DOM directly, by updating the domNode.value. The issue here is then the state & the value in the UI are out of sync. Think about you update props then to the old value. The virtual dom will think it never changed and won't update the component although it should. Do you know what I mean?